framework/decode/vulkan_address_replacer.cpp

```diff
@@ -0,0 +1,791 @@
+        physical_device_info->replay_device_info->raytracing_properties)
+    {
+        capture_ray_properties_ = *physical_device_info->capture_raytracing_properties;
+        replay_ray_properties_  = *physical_device_info->replay_device_info->raytracing_properties;
+
+        if (capture_ray_properties_.shaderGroupHandleSize != replay_ray_properties_.shaderGroupHandleSize ||
+            capture_ray_properties_.shaderGroupHandleAlignment != replay_ray_properties_.shaderGroupHandleAlignment ||
+            capture_ray_properties_.shaderGroupBaseAlignment != replay_ray_properties_.shaderGroupBaseAlignment)
+        {
+            valid_sbt_alignment_ = false;
+        }
+
+        GFXRECON_ASSERT(physical_device_info->replay_device_info != nullptr);
+        GFXRECON_ASSERT(physical_device_info->replay_device_info->memory_properties.has_value());
+        memory_properties_ = *physical_device_info->replay_device_info->memory_properties;
+    }
+}
+
+VulkanAddressReplacer::VulkanAddressReplacer(VulkanAddressReplacer&& other) noexcept : VulkanAddressReplacer()
+{
+    swap(*this, other);
+}
+
+VulkanAddressReplacer::~VulkanAddressReplacer()
+{
+    mark_injected_commands_helper_t mark_injected_commands_helper;
+
+    // explicitly free resources here, in order to mark destruction API-calls as injected
+    pipeline_context_sbt_ = {};
+    pipeline_context_bda_ = {};
+    shadow_sbt_map_       = {};
+
+    if (pipeline_bda_ != VK_NULL_HANDLE)
+    {
+        device_table_->DestroyPipeline(device_, pipeline_bda_, nullptr);
+    }
+    if (pipeline_sbt_ != VK_NULL_HANDLE)
+    {
+        device_table_->DestroyPipeline(device_, pipeline_sbt_, nullptr);
+    }
+    if (pipeline_layout_ != VK_NULL_HANDLE)
+    {
+        device_table_->DestroyPipelineLayout(device_, pipeline_layout_, nullptr);
+    }
+}
+
+void VulkanAddressReplacer::ProcessCmdTraceRays(
+    const VulkanCommandBufferInfo*                                                              command_buffer_info,
+    VkStridedDeviceAddressRegionKHR*                                                            raygen_sbt,
+    VkStridedDeviceAddressRegionKHR*                                                            miss_sbt,
+    VkStridedDeviceAddressRegionKHR*                                                            hit_sbt,
+    VkStridedDeviceAddressRegionKHR*                                                            callable_sbt,
+    const decode::VulkanDeviceAddressTracker&                                                   address_tracker,
+    const std::unordered_map<graphics::shader_group_handle_t, graphics::shader_group_handle_t>& group_handle_map)
+{
+    GFXRECON_ASSERT(device_table_ != nullptr);
+
+    // NOTE: we expect this map to be populated here, but not for older captures (before #1844) using trimming.
+    if (group_handle_map.empty())
+    {
+        // the capture appears to be older and is missing information we require here -> bail out
```

Comment:
Maybe log an `INFO_ONCE` level message here?  (Side note, I don't even know if we have an INFO_ONCE, if not, just use INFO).

Reply:
I'll check. and yes, was also thinking back/forth to use warning or just INFO

Reply:
we do have a `GFXRECON_LOG_INFO_ONCE`, using that

Reply:
done